services/frontend-service/src/ui/Pages/Home/Home.test.tsx

```diff
@@ -32,20 +32,278 @@ describe('App', () => {
+        expect(mock_ServiceLane.ServiceLane.getCallArgument(0, 0)).toStrictEqual({ application: appBuilder('1') });
+        expect(mock_ServiceLane.ServiceLane.getCallArgument(1, 0)).toStrictEqual({ application: appBuilder('2') });
+        expect(mock_ServiceLane.ServiceLane.getCallArgument(2, 0)).toStrictEqual({ application: appBuilder('3') });
+    });
+});
+
+describe('Application Filter', () => {
+    interface dataT {
+        name: string;
+        query: string;
+        applications: string[];
+        expect: (nrLocks: number) => void;
+    }
+
+    const data: dataT[] = [
+        {
+            name: 'filter applications - 1 result',
+            applications: ['dummy', 'test', 'test2', 'foo'],
+            query: 'dummy',
+            expect: (nrLocks) => expect(nrLocks).toStrictEqual(1),
+        },
+        {
+            name: 'filter applications - 0 results',
+            applications: ['dummy', 'test', 'test2'],
+            query: 'foo',
+            expect: (nrLocks) => expect(nrLocks).toStrictEqual(0),
+        },
+        {
+            name: 'filter applications - 2 results',
+            applications: ['dummy', 'test', 'test2'],
+            query: 'test',
+            expect: (nrLocks) => expect(nrLocks).toStrictEqual(2),
+        },
+    ];
+
+    describe.each(data)(`Renders an Application Card`, (testcase) => {
+        it(testcase.name, () => {
+            // when
+            const nrLocks = testcase.applications.filter((val) => searchCustomFilter(testcase.query, val)).length;
+            testcase.expect(nrLocks);
+        });
+    });
+});
+
+describe('Get teams from application list (useTeamNames)', () => {
+    interface dataT {
+        name: string;
+        applications: { [key: string]: Application };
+        expect: (teamNames: string[]) => void;
+    }
+
+    const data: dataT[] = [
+        {
+            name: 'right amount of teams - 4 sorted results',
+            applications: {
+                foo: {
+                    name: 'foo',
+                    releases: [],
+                    sourceRepoUrl: 'http://foo.com',
+                    team: 'dummy',
+                },
+                bar: {
+                    name: 'bar',
+                    releases: [],
+                    sourceRepoUrl: 'http://bar.com',
+                    team: 'test',
+                },
+                example: {
+                    name: 'example',
+                    releases: [],
+                    sourceRepoUrl: 'http://example.com',
+                    team: 'test2',
+                },
+                team: {
+                    name: 'team',
+                    releases: [],
+                    sourceRepoUrl: 'http://team.com',
+                    team: 'foo',
+                },
+            },
+            expect: (teamNames) => expect(teamNames).toStrictEqual(['dummy', 'foo', 'test', 'test2']),
```

Comment:
Instead of having the function on the expect, modify the expect so that it is a number value. And then in the test itself expect the value in the testcase

Reply:
rewrote tests to fit the table driven tests paradigm as mentioned here: https://github.com/freiheit-com/kuberpult/pull/380#discussion_r997927036